src/odemis/driver/semnidaq.py

```diff
@@ -1326,40 +1348,156 @@ def _acquire_frames(self,
+                try:
+                    new_samples_n = ci_reader.read_many_sample_uint32(ci_buffer[:samples_to_acquire],
+                                                                      number_of_samples_per_channel=samples_to_acquire,
+                                                                      timeout=0.1)  # Should be immediate as we just checked the period
+                except nidaqmx.DaqReadError as ex:
+                    raise IOError("Failed to read CI acquisition data") from ex
+
+                if new_samples_n != samples_to_acquire:
+                    logging.warning("Only received %d samples, while expecting %d, will try more",
+                                    new_samples_n, samples_to_acquire)
+
+                logging.debug("Got another %s CI samples, over %s still to acquire", new_samples_n, samples_left_n)
+                logging.debug(
+                    f"ci_data[c] shape = {ci_data[c].shape}, ci_buffer shape: {ci_buffer[:new_samples_n].shape}")
+                ci_prev_samples_n[c], ci_prev_samples_sum[c] = self._downsample_data(ci_data[c],
+                                                                                     acq_settings.res,
+                                                                                     acq_settings.margin,
+                                                                                     ci_acquired_n,
+                                                                                     acq_settings.ao_osr,
+                                                                                     ci_buffer[:new_samples_n],
+                                                                                     ci_prev_samples_n[c],
+                                                                                     ci_prev_samples_sum[c],
+                                                                                     acc_dtype=ci_data.dtype,
+                                                                                     # for sum, this is the same as data.dtype
+                                                                                     average=False)
+            ci_acquired_n += new_samples_n
+
+        return ci_acquired_n, ci_prev_samples_n, ci_prev_samples_sum
+
     def _configure_sync_tasks(self,
                               acq_settings: AcquisitionSettings,
                               ao_task: nidaqmx.Task, do_task: nidaqmx.Task, ai_task: nidaqmx.Task,
+                              ci_tasks: List[nidaqmx.Task],
                               ao_buffer_n: int, do_buffer_n: int, ai_buffer_n: int,
                               ):
         """
         Configures the synchronization tasks for AI, AO, and DO.
         :param ao_task: The AO task object.
         :param do_task: The DO task object.
         :param ai_task: The AI task object.
+        :param ci_tasks: a list of counter input tasks.
```

Comment:
```suggestion
        :param ci_tasks: The list of counter input tasks.
```

not much of a change, but it is just triggering my OCD :smile: